### PR TITLE
Add custom image downloader - reject responses with large content length

### PR DESCRIPTION
**Pull request by OlegAba**
Ref: #310 conversation

Added a `20MB` content length limit for all image files

### CODE REVIEW

**Comment by OlegAba** [comment]
Large PFP and Banner Image(172MB):
`npub1yqqwsyt595pwtgf7sax7kgfl2z8fpewc0jmjaajnv6hgfxhckutq7ydzdd`

Before             |  After
:-------------------------:|:-------------------------:
 |

File: damus/Models/KFImageModel.swift [hunk 1]
```diff
@@ -106,3 +108,28 @@ struct CustomCacheSerializer: CacheSerializer {
         return DefaultCacheSerializer.default.image(with: data, options: options)
     }
 }
+
+class CustomSessionDelegate: SessionDelegate {
```
Comment (jb55) [inline damus/Models/KFImageModel.swift:112]: I'm confused as to where the HEAD request is happening here

Comment (OlegAba) [reply damus/Models/KFImageModel.swift:112]: No separate request is being made. The HEAD is being intercepted in the urlSession response which runs before the download starts in KF. I think this is more efficient and cleaner than running a separate HEAD request and then passing the url to KF.

Comment (jb55) [reply damus/Models/KFImageModel.swift:112]: interesting...

Comment (OlegAba) [reply damus/Models/KFImageModel.swift:112]: Let me know if you have any other questions or prefer it to be done another way. Also, looks like there are merge conflicts now. I can resolve those once everything looks good.

Comment (jb55) [reply damus/Models/KFImageModel.swift:112]: I'm just curious how you know this header is available before the data is downloaded, is there docs for this somewhere?

Comment (jb55) [reply damus/Models/KFImageModel.swift:112]: ah awesome

**Comment by OlegAba** [comment]
@jb55 Sorry for adding so many changes for a simple rebase but I think it was needed.  The code is much cleaner now. 

Not sure why the tests are failing. It appears to be unrelated to the changes?

**Comment by jb55** [comment]
the option setter thing is long overdue, thank you

**Pull request closed without merging**